Use an env variable to roll out model sharding cache

hstonec opened the pull request:
Summary: We got positive result from AI Labs for the cache but would like to quantify the gains in produciton. As torchrec is an open source lib, we cannot use JK so need to rely on an environment variable to control it.

Reviewed By: andywag

Differential Revision: D70078729

andywag reviewed (approved):
Review automatically exported from Phabricator review in Meta.

hstonec committed:
Use an env variable to roll out model sharding cache (#2764)

Summary:

We got positive result from AI Labs for the cache implemented in D69443708 and D68578829 but would like to quantify the gains in produciton. As torchrec is an open source lib, we cannot use JK so need to rely on an environment variable to control it.

Differential Revision: D70078729

andywag reviewed (changes requested):
Review automatically exported from Phabricator review in Meta.
